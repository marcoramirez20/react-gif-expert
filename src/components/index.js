
/*
* Esto se conoce como un archivo de barril o archivo indice
* */

export * from './AddCategory.jsx'
export * from './GifGrid.jsx'
export * from './GifItem.jsx'