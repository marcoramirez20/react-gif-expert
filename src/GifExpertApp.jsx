import {useState} from 'react';
import {AddCategory, GifGrid} from './components';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball'])

    const onNewCategory = (category) => {
        if(categories.includes(category)){
            return;
        }
        setCategories(cat => { return [category, ...cat] })
    }

    return(
        <>
            <h1>GifExpertApp</h1>
            <AddCategory
                onNewCategory={onNewCategory}
            />

            {categories.map( category => (
                  <GifGrid
                      key={category}
                      category={category}
                  />
              )
            )}
        </>
    )
}