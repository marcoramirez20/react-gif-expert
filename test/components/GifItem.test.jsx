import {render, screen} from "@testing-library/react";
import {GifItem} from "../../src/components";

describe('Tests of GifItem', () => {

    const title = 'Dragon Ball'
    const url = 'http://dragon.ball/goku.jpg'

    test('Should match with snapshot', () => {
        const {container} = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('Should be exists image source and alternative text', () => {
        render(<GifItem title={title} url={url} />)
        // screen.debug()
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)
    })

    test('Should show the title in the component', () => {
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText).toBeTruthy()
    })
});