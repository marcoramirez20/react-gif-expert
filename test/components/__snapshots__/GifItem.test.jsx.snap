// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of GifItem Should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="http://dragon.ball/goku.jpg"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
